ext/sqlite3/exception.c

```diff
@@ -115,3 +115,116 @@ rb_sqlite3_raise_msg(sqlite3 *db, int status, const char *msg)
     rb_iv_set(exception, "@code", INT2FIX(status));
     rb_exc_raise(exception);
 }
+
+void
+rb_sqlite3_raise_with_sql(sqlite3 *db, int status, const char *sql)
+{
+    VALUE klass = Qnil;
+    VALUE error_message = Qnil;
+    const char* sqlite_error_msg = sqlite3_errmsg(db);
+    int error_offset = sqlite3_error_offset(db);
+
+    /* Consider only lower 8 bits, to work correctly when
+       extended result codes are enabled. */
+    switch (status & 0xff) {
```

Comment:
This should probably be exctraced in a function rather than copy pasted.

Reply:
I strongly agree.